### PR TITLE
refactor(legal): move legal pages to frontend, simplify `ui`

### PR DESCRIPTION
**Pull request by elbotho**
- remove legal pages code
- redirect to new frontend urls (`/legal` instead of incorrect `/imprint`)
- remove preact dependency (html template strings are enough for the simple use cases here imo)
- remove other now unused dependencies
- redirect to frontend when meeting room redirect is not found
- no external assets needed for `ui` any more
- redirect code are now explicit (was needed for the type somehow but I like it anyway)

~running `yarn` after removing the packages broke the tests but I'm pretty sure that is not related to the code changes 🤔~
before the unexpected (probably unrelated) type problem this was a really clean PR 😬 @kulla did you encounter problems with the `RequestInit` or fetch types before?

### CODE REVIEW

File: src/frontend-proxy.ts [hunk 1]
```diff
@@ -59,9 +59,12 @@ async function fetchBackend({
     backendUrl.pathname = backendUrl.pathnameWithoutTrailingSlash
   }
 
-  const response = await fetch(new Request(backendUrl.toString(), request), {
-    redirect: route.__typename === 'Frontend' ? route.redirect : 'manual',
-  })
+  const response = await fetch(
+    new Request(backendUrl.toString(), { headers: request.headers }),
```
Comment (elbotho) [inline src/frontend-proxy.ts]: Not sure about this one. Is it enough to pass the headers?

File: __tests__/utils.tsx [hunk 2]
```diff
@@ -208,31 +204,8 @@ describe('isInstance()', () => {
...
-    ['* 1\n* 2', '<ul>\n<li>1</li>\n<li>2</li>\n</ul>'],
-    ['', ''],
-  ])('Markdown: %p', (markdown, html) => {
-    expect(markdownToHtml(markdown)).toBe(html)
-  })
-})
-
-test('PreactResponse', async () => {
-  const hello = createPreactResponse(<h1>Hello</h1>)
+test('HtmlResponse', async () => {
```
Comment (kulla) [inline __tests__/utils.tsx:207]: Follow-Up: Should not be needed... I guess this can be deleted as well...

File: jest.setup.ts [hunk 3]
```diff
@@ -48,19 +48,14 @@ afterAll(() => {
...
-          cryptoNode
-            .createHash(encoding.toLowerCase().replace('-', ''))
-            .update(message)
-            .digest(),
-        )
-      },
-    },
-    randomUUID: cryptoNode.randomUUID,
-  } as unknown as typeof crypto
+  globalThis.cryptoDigest = (encoding: string, message: Uint8Array) => {
```
Comment (kulla) [inline jest.setup.ts]: I think this should be `crypo.subtle` -> See https://developers.cloudflare.com/workers/runtime-apis/web-crypto#subtlecrypto-methods

File: jest.setup.ts [hunk 4]
```diff
@@ -88,4 +83,6 @@ declare global {
   var server: ReturnType<typeof import('msw/node').setupServer>
   // eslint-disable-next-line no-var
   var sentryEvents: SentryEvent[]
+  // eslint-disable-next-line no-var
+  var cryptoDigest: (encoding: string, message: Uint8Array) => Promise<Buffer>
```
Comment (kulla) [inline jest.setup.ts]:
```suggestion
```

There is no global `cryptoDigest()` in CF worker -> So this will not work...

**Review by kulla — CHANGES_REQUESTED** [review]

**Review by kulla — APPROVED** [review]
Okay, I have fixed all leftover issues. It should be save to deploy...

**Comment by kulla** [comment]
> before the unexpected (probably unrelated) type problem this was a really clean PR 😬 @kulla did you encounter problems with the `RequestInit` or fetch types before?

This is due to https://github.com/cloudflare/workerd/issues/1298 I have fixed the type errors...

File: __tests__/__utils__/test-environment.ts [hunk 5]
```diff
@@ -63,8 +64,8 @@ export abstract class TestEnvironment {
 
   public abstract fetchRequest(request: Request): Promise<Response>
 
-  public createRequest(spec: UrlSpec, init?: RequestInit) {
-    return new Request(this.createUrl(spec), init)
+  public createRequest(spec: UrlSpec, init?: RequestInit<CfProperties>) {
+    return new Request<unknown, CfProperties>(this.createUrl(spec), init)
```
Comment (elbotho) [inline __tests__/__utils__/test-environment.ts:68]: nice!

**Pull request merged**